data/dynos_bin_goddard.cpp

```
@@ -0,0 +1,149 @@
+    printf("[DynOS] Goddard: loaded mario_head.gdbin (%d bytes) from %s\n", _Size, _Path.c_str());
+}
+
+SysPath Goddard_CalculateActiveMarioHeadBin() {
+    // Get the character index from save file
+    u32 charIndex = (configPlayerModel >= CT_MAX) ? CT_MARIO : configPlayerModel;
+    
+    // First try to find a character-specific head, then fall back to mario_head
+    for (auto& _Pack : DynosPacks()) {
+        if (!_Pack.mEnabled) { continue; }
+        
+        // Try character-specific head first
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)